### PR TITLE
Searchnav

### LINKED ISSUE #437: Hub- fix Play All on releases

### PR DESCRIPTION
**Pull request by ikjudd**
closes #437 and #438

### CODE REVIEW

File: js/web/components/Hub.js [hunk 1]
```diff
@@ -1,11 +1,10 @@
 import { useEffect, useMemo, useState, useContext } from 'react'
 import dynamic from 'next/dynamic'
-import Head from 'next/head'
 import { Box, Typography } from '@mui/material'
 import { styled } from '@mui/system'
 import Hub from '@nina-protocol/nina-internal-sdk/esm/Hub'
 import Release from '@nina-protocol/nina-internal-sdk/esm/Release'
-
+import Head from 'next/head'
```
Comment (ejf89) [inline js/web/components/Hub.js:7]: remove unused import plz

File: js/web/components/NavSearch.js [hunk 2]
```diff
@@ -0,0 +1,397 @@
...
+      e.preventDefault()
+      e.stopPropagation()
+      return
+    }
+    setShowDropdown(false)
+    setShowSearchInput(false)
+  }
+
+  const autoCompleteHandler = async (query) => {
+    const response = await axios.post(
```
Comment (ejf89) [inline js/web/components/NavSearch.js:92]: lets add the SDK functon for this

File: js/web/components/Release.js [hunk 3]
```diff
@@ -28,6 +28,13 @@ const ReleaseComponent = ({ metadataSsr }) => {
     metadataSsr || releaseState?.metadata[releasePubkey] || null
   )
   const release = useMemo(() => releaseState.tokenData[releasePubkey], [releaseState, releasePubkey])
+  console.log('release', release)
+  console.log(releaseState)
```
Comment (ejf89) [inline js/web/components/Release.js:32]: remove logs plx

File: js/web/components/ReleasePurchase.js [hunk 4]
```diff
@@ -21,7 +21,7 @@ import rehypeExternalLinks from "rehype-external-links";
 import Royalty from './Royalty'
 
 const ReleasePurchase = (props) => {
-  const { releasePubkey, metadata, router } = props
+  const { releasePubkey, metadata, router, } = props
```
Comment (ejf89) [inline js/web/components/ReleasePurchase.js:24]: trailing comma

File: js/web/components/Release.js [hunk 5]
```diff
@@ -28,6 +28,13 @@ const ReleaseComponent = ({ metadataSsr }) => {
...
+  useEffect(() => {
+    if (releasePubkey) {
+      getExchangeHistoryForRelease(releasePubkey)
```
Comment (ejf89) [inline js/web/components/Release.js:35]: release page is broken, may be related to this

File: js/web/components/ReusableTable.js [hunk 6]
```diff
@@ -104,7 +136,7 @@ const HubDescription = ({ description }) => {
   )
 }
 
-const ReusableTableBody = ({ items, tableType }) => {
+const ReusableTableBody = ({ items, tableType, hasOverflow }) => {
```
Comment (ejf89) [inline js/web/components/ReusableTable.js:139]: +1

File: js/web/components/Search.js [hunk 7]
```diff
@@ -0,0 +1,243 @@
...
+    if (searchFilter === 'hubs') {
+      setActiveView(3)
+    }
+  }, [searchFilter])
+
+  const renderTables = (activeView) => {
+    switch (activeView) {
+      case 1:
+        return (
+          <ResultsWrapper>
```
Comment (ejf89) [inline js/web/components/Search.js:61]: `ResultsWrapper` can be refactored and written once around this function

File: js/web/components/SearchDropdown.js [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+      )}    
+    </>
+  )
+}
+
+const SearchResultsWrapper = styled(Box)(({theme}) => ({
+  display: 'flex',
+  flexDirection: 'column',
+  '&:focus': {
+    backgroundColor: 'rgba(255,255,255,0.8)',
```
Comment (ejf89) [inline js/web/components/SearchDropdown.js:88]: transparency color values should be added to / pulled from the theme object

**Review by ejf89 — CHANGES_REQUESTED** [review]
almost there, needs some small tweaks

**Comment by facing-n** [comment]
Screen Shot 2022-10-18 at 10 24 21 AM

Releases in search Results should be `{Artist} - {Title}`, here they are just `{Title}`

File: js/web/components/Hub.js [hunk 9]
```diff
@@ -1,11 +1,10 @@
 import { useEffect, useMemo, useState, useContext } from 'react'
 import dynamic from 'next/dynamic'
-import Head from 'next/head'
-import { Box, Typography } from '@mui/material'
+import { Box, } from '@mui/material'
 import { styled } from '@mui/system'
 import Hub from '@nina-protocol/nina-internal-sdk/esm/Hub'
 import Release from '@nina-protocol/nina-internal-sdk/esm/Release'
-
+import Head from 'next/head'
```
Comment (ejf89) [inline js/web/components/Hub.js:7]: `Head` and associated mark up should be in the associated files in `/pages`

File: js/web/components/NavSearch.js [hunk 10]
```diff
@@ -0,0 +1,398 @@
...
+  const dropdownRef = useRef()
+  const searchInputRef = useRef()
+
+  useEffect(() => {
+    NinaSdk.client.init(
+      process.env.NINA_API_ENDPOINT,
+      process.env.SOLANA_CLUSTER_URL,
+      process.env.NINA_PROGRAM_ID
+    )
+    console.log('NinaSdk.client', NinaSdk.client)
```
Comment (ejf89) [inline js/web/components/NavSearch.js:35]: kill log

File: js/web/components/Profile.js [hunk 11]
```diff
@@ -54,18 +54,18 @@ const Profile = ({ profilePubkey }) => {
   useEffect(() => {
     const getUserData = async (profilePubkey) => {
       const hubs = await getHubsForUser(profilePubkey)
-      
+
       const [collected, published] = await getUserCollectionAndPublished(
         profilePubkey
       )
-
+      console.log('collected: ', collected)
```
Comment (ejf89) [inline js/web/components/Profile.js:61]: kill log

File: js/web/components/ReusableTable.js [hunk 12]
```diff
@@ -197,6 +233,33 @@ const ReusableTableBody = ({ items, tableType }) => {
...
+      formattedData = {
+        link: `/profiles/${data?.publicKey}`,
+        collaborator: data.name
+      }
+    }
+
+    if(tableType === 'searchResultReleases' || tableType === 'filteredSearchResultReleases'){
+      formattedData= {
+        id: data?.publicKey,
+        // ctas: data?.publicKey,
```
Comment (ejf89) [inline js/web/components/ReusableTable.js:247]: remove unused vars

File: js/web/components/ReusableTable.js [hunk 13]
```diff
@@ -197,6 +233,33 @@ const ReusableTableBody = ({ items, tableType }) => {
...
+        image: data?.image,
+        link: `/${data?.publicKey}`,
+        title: data.title,
+      }
+    }
+    if(tableType === 'searchResultHubs' || tableType === 'filteredSearchResultHubs'){
+      formattedData ={
+        // ctas: data?.publicKey,
```
Comment (ejf89) [inline js/web/components/ReusableTable.js:255]: same as above

File: js/web/components/Hub.js [hunk 14]
```diff
@@ -69,10 +67,12 @@ const HubComponent = ({ hubHandle, hubPubkey }) => {
 
     const data = hubReleases?.map((hubRelease) => {
       const releaseMetadata = releaseState.metadata[hubRelease.release]
+      releaseMetadata.authority = releaseState.tokenData[hubRelease.release].authority
       releaseMetadata.releasePubkey = hubRelease.release
       return releaseMetadata
     })
     setReleaseData(data)
+    console.log('data', data)
```
Comment (ejf89) [inline js/web/components/Hub.js:75]: log

**Review by ejf89 — APPROVED** [review]

**Review by ejf89 — COMMENTED** [review]
waiting for fix in search

**Pull request closed without merging**